### PR TITLE
Moving all errors to Errors class

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://github.com/demisto/etc/issues/24232

## Description
Moving all errors to Errors class

### CODE REVIEW

File: demisto_sdk/commands/common/constants.py [hunk 1]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+
+    @staticmethod
+    def description_in_package_and_yml(package_path):
+        return f"DS103 - A description was found both in the " \
+               f"package and in the yml, please update the package {package_path}."
+
+    @staticmethod
+    def default_docker_error():
+        return 'DO100 - The current docker image in the yml file is the default one: demisto/python:1.3-alpine,\n' \
+               'Please create or use another docker image\n'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:314]: Maybe we should add the option to skip this check if they really want to use it?

File: demisto_sdk/commands/common/hook_validations/docker.py [hunk 2]
```diff
@@ -204,7 +202,7 @@ def get_docker_image_latest_tag(docker_image_name, yml_docker_image):
                 print_warning('docker image must be a demisto docker image. When the docker image is ready,'
                               ' please rename it to: demisto/<image>:<tag>')
```
Comment (ronykoz) [inline demisto_sdk/commands/common/hook_validations/docker.py:203]: let's add this one too

File: demisto_sdk/commands/common/constants.py [hunk 3]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+    def not_demisto_docker():
+        return 'DO102 - docker image must be a demisto docker image. e.g: demisto/<image>:<tag>'
+
+    @staticmethod
+    def docker_tag_not_fetched(docker_image_name):
+        return f'DO103 - Failed getting tag for: {docker_image_name}. Please check it exists and of demisto format.'
+
+    @staticmethod
+    def no_docker_tag():
+        return 'DO104 - The docker image in your integration/script does not have a tag, please attach the latest tag'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:333]: take the explanation from  - we don't want `latest`
f'DO101 - "latest" tag is not allowed,\n' \
               f'Please create or update to an updated versioned image\n' \
               f'You can check for the most updated version of {docker_image_tag} ' \
               f'here: https://hub.docker.com/r/{docker_image_name}/tags\n'

File: demisto_sdk/commands/common/constants.py [hunk 4]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+
+    @staticmethod
+    def docker_not_formatted_correctly(docker_image):
+        return f'DO105 - The docker image: {docker_image} is not of format - demisto/image_name'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:337]:
```suggestion
        return f'DO105 - The docker image: {docker_image} is not of format - demisto/image_name:X.X'
```

File: demisto_sdk/commands/common/constants.py [hunk 5]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+        return f'IF107 - {file_path}: fromVersion must be at least 5.0.0'
+
+    @staticmethod
+    def invalid_incident_field_or_type_from_version(file_path):
+        return f'IF108 - {file_path}: "fromVersion" has an invalid value.'
+
+    @staticmethod
+    def new_incident_field_required(file_path):
+        return f'IF109 - {file_path}: new incident fields can not be required. change to:\nrequired: false.'
+
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:424]: you've skipped 110

File: demisto_sdk/commands/common/constants.py [hunk 6]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+    @staticmethod
+    def incident_field_type_change(file_path):
+        return f'IF111 - {file_path}: Changing incident field type is not allowed.'
+
+    @staticmethod
+    def incident_type_integer_field(file_path, field):
+        return f'IF112 - {file_path}: the field {field} needs to be a positive integer. Please add it.\n'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:431]:
```suggestion
        return f'IT112 - {file_path}: the field {field} needs to be a positive integer. Please add it.\n'
```

File: demisto_sdk/commands/common/constants.py [hunk 7]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+
+    @staticmethod
+    def invalid_context_output(command_name, output):
+        return f'IN120 - Invalid context output for command {command_name}. Output is {output}'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:435]: CTX

File: demisto_sdk/commands/common/constants.py [hunk 8]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+        return F'RN101 - No new comment has been added in the release notes file: {release_notes_path}'
+
+    @staticmethod
+    def release_notes_not_formatted_correctly(release_notes_path, link_to_rn_standard):
+        return F'RN102 - File {release_notes_path} is not formatted according to ' \
+               F'release notes standards.\nFix according to {link_to_rn_standard}'
+
+    @staticmethod
+    def pack_file_does_not_exist(file_name):
+        return f'"PU100 - {file_name}" file does not exist, create one in the root of the pack'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:459]: PU ? 
let's do P

File: demisto_sdk/commands/common/constants.py [hunk 9]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+    def pack_metadata_name_not_valid():
+        return f'PU108 - Pack metadata {PACK_METADATA_NAME} field is not valid. Please fill valid pack name.'
+
+    @staticmethod
+    def pack_metadata_empty():
+        return 'PU105 - Pack metadata is empty.'
+
+    @staticmethod
+    def pack_metadata_should_be_dict():
+        return 'PU106 - Pack metadata should be a dictionary.'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:487]: which pack?

File: demisto_sdk/commands/common/constants.py [hunk 10]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+    def empty_field_in_pack_metadata(list_field):
+        return f'PU111 - Empty value in the {list_field} field.'
+
+    @staticmethod
+    def pack_metadata_isnt_json(pack_meta_file):
+        return f'PU112 - Could not parse {pack_meta_file} file contents to json format'
+
+    @staticmethod
+    def playbook_cant_have_rolename():
+        return "PB100 - Playbook can not have a rolename."
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:511]: which PB?

File: demisto_sdk/commands/common/constants.py [hunk 11]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+
+    @staticmethod
+    def playbook_unreachable_condition(task_id, next_task_branch):
+        return f'PB101 - Playbook conditional task with id:{task_id} has task with unreachable ' \
+               f'next task condition "{next_task_branch}". Please remove this task or add ' \
+               f'this condition to condition task with id:{task_id}.'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:517]: which PB?

File: demisto_sdk/commands/common/constants.py [hunk 12]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+
+    @staticmethod
+    def playbook_unhandled_condition(task_id, task_condition_labels):
+        return f'PB102 - Playbook conditional task with id:{task_id} has unhandled ' \
+               f'condition: {",".join(map(lambda x: f"{str(x)}", task_condition_labels))}'
+
+    @staticmethod
+    def playbook_unconnected_tasks(orphan_tasks):
+        return f'PB103 - The following tasks ids have no previous tasks: {orphan_tasks}'
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:526]: which pb

**Comment by Itay4** [comment]
lets see if we can start getting rid of `print_error` and `print_warning` and use `click.echo` instead

File: demisto_sdk/commands/common/constants.py [hunk 13]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
                f" e.g: DBotTrainTextClassifierV2"
 
     @staticmethod
     def found_hidden_param(parameter_name):
-        return f"Parameter: \"{parameter_name}\" can't be hidden. Please remove this field."
+        return f"IN130 - Parameter: \"{parameter_name}\" can't be hidden. Please remove this field."
+
+    @staticmethod
+    def id_should_equal_name(name, file_id, file_path):
+        return "BA101 - The File's name, which is: '{}', should be equal to its ID, which is: '{}'."\
```
[the remark this answers is not in the document]
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:240]: BAsic - also thought of GE for general

File: demisto_sdk/commands/common/constants.py [hunk 14]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+    def test_playbook_not_configured(content_item_id, missing_test_playbook_configurations,
+                                     missing_integration_configurations):
+        return f'TP100 - The TestPlaybook {content_item_id} is not registered in {CONF_PATH} file.\n ' \
+               f'Please add\n{missing_test_playbook_configurations}\n ' \
+               f'or if this test playbook is for an integration\n{missing_integration_configurations}\n ' \
+               f'to {CONF_PATH} path under \'tests\' key.'
+
+    @staticmethod
+    def integration_not_registered(file_path, missing_test_playbook_configurations, no_tests_key):
+        return f'IN100 - The following integration is not registered in {CONF_PATH} file.\n' \
```
[the remark this answers is not in the document]
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:253]: as in CJ = conf json?

File: demisto_sdk/commands/common/constants.py (hunk 1 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:315]: isnt that what the ignore should do?

File: demisto_sdk/commands/common/constants.py (hunk 3 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py (hunk 4 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py (hunk 5 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:426]: did not, see: from_version_modified_after_rename

File: demisto_sdk/commands/common/constants.py (hunk 6 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:433]: I can change that (that was the original plan but this is the only incident-type specific error - the rest are with incident-fields so I kept them together

File: demisto_sdk/commands/common/constants.py (hunk 8 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:461]: Pack Unique - I think we should stick to 2 letters for consistency.

File: demisto_sdk/commands/common/constants.py (hunk 7 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:437]: there are only two errors connected to context (which are not dbot score nor BC related) - do we want to create a section for those?

File: demisto_sdk/commands/common/constants.py (hunk 9 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done (in all related)

File: demisto_sdk/commands/common/constants.py (hunk 10 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py (hunk 11 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py (hunk 12 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py [hunk 15]
```diff
@@ -153,19 +154,15 @@ def id_changed(file_path):
...
+        return "ST101 - File's ID contains slashes - please remove."
 
     @staticmethod
     def missing_release_notes(file_path, rn_path):
-        return '{}:  is missing releaseNotes, Please add it under {}'.format(file_path, rn_path)
-
-    @staticmethod
-    def display_param(param_name, param_display):
-        return 'The display name of the {} parameter should be \'{}\''.format(param_name, param_display)
+        return 'RN100 - {}:  is missing releaseNotes, Please add it under {}'.format(file_path, rn_path)
```
[the remark this answers is not in the document]
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py [hunk 16]
```diff
@@ -221,21 +219,344 @@ def no_common_server_python(path):
...
+        return f'TP101 - You don\'t have a TestPlaybook for {file_type} {file_path}. ' \
+               f'If you have a TestPlaybook for this {file_type}, ' \
+               f'please edit the yml file and add the TestPlaybook under the \'tests\' key. ' \
+               f'If you don\'t want to create a TestPlaybook for this {file_type}, ' \
+               f'edit the yml file and add  \ntests:\n -  No tests\n lines to it or ' \
+               f'run \'demisto-sdk format -i {file_path}\''
+
+    @staticmethod
+    def test_not_in_conf_json(file_id):
+        return f"CJ101 - You've failed to add the {file_id} to conf.json"
```
[the remark this answers is not in the document]
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

**Comment by roysagi** [comment]
> lets see if we can start getting rid of `print_error` and `print_warning` and use `click.echo` instead

This is just the first change in the process - in order to get all the errors in a single location. I want to create a function that handles errors and is able to ignore them - thus all prints will be done in a single location - there I'll see if I can make echo work.

File: demisto_sdk/commands/common/constants.py [hunk 17]
```diff
@@ -10,56 +10,45 @@ def suggest_fix(file_path: str, *args: Any, cmd: str = 'format') -> str:
...
-    def dbot_invalid_description(file_path, command_name, missing_descriptions, context_standard):
-        return "DB101 - {}: The DBotScore description of the reputation command {} aren't valid. Missing: {}. " \
-               "Fix according to context standard {} " \
-            .format(file_path, command_name, missing_descriptions, context_standard)
+    def no_default_arg(file_path, command_name):
+        return "IN107 - {}: Could not find default argument " \
+               "{} in command {}".format(file_path, command_name, command_name)
 
     @staticmethod
     def missing_reputation(file_path, command_name, reputation_output, context_standard):
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:54]: 8 9 10 ?

File: demisto_sdk/commands/common/constants.py [hunk 18]
```diff
@@ -72,11 +61,6 @@ def wrong_subtype(file_name):
         return "IN111 - {}: The subtype for our yml files should be either python2 or python3, " \
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py]: 12?

File: demisto_sdk/commands/common/constants.py [hunk 19]
```diff
@@ -98,22 +92,69 @@ def duplicate_param(param_name, file_path):
         return "IN118 - {}: The parameter '{}' of the " \
                "file is duplicated, please remove one of its appearances.".format(file_path, param_name)
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:94]: 119?

File: demisto_sdk/commands/common/constants.py [hunk 20]
```diff
@@ -98,22 +92,69 @@ def duplicate_param(param_name, file_path):
...
 
+    @staticmethod
+    def invalid_context_output(command_name, output):
+        return f'IN120 - Invalid context output for command {command_name}. Output is {output}'
+
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:98]: 121?

File: demisto_sdk/commands/common/constants.py [hunk 21]
```diff
@@ -98,22 +92,69 @@ def duplicate_param(param_name, file_path):
...
+               f"e.g: Kenna v2, Jira v2"
+
+    @staticmethod
+    def found_hidden_param(parameter_name):
+        return f"IN130 - Parameter: \"{parameter_name}\" can't be hidden. Please remove this field."
+
+    @staticmethod
+    def invalid_v2_script_name(file_path):
+        return f"SC103 - The name of the v2 script : {file_path} is incorrect , should be **name**V2." \
+               f" e.g: DBotTrainTextClassifierV2"
```
Comment (ronykoz) [inline demisto_sdk/commands/common/constants.py:145]:
```suggestion
        return f"SC100 - The name of the v2 script : {file_path} is incorrect , should be **name**V2." \
               f" e.g: DBotTrainTextClassifierV2"
```

Comment (roysagi) [reply demisto_sdk/commands/common/constants.py]: done

File: demisto_sdk/commands/common/constants.py (hunk 6 above)
Comment (roysagi) [reply demisto_sdk/commands/common/constants.py:422]: done

**Comment by roysagi** [comment]
> you have a bunch with no error code

Right these are general error - some of them seem to be unused and some are used for lint so I did not give them an error code.

**Comment by roysagi** [comment]
closing this pr as it is covered by the full error handling pr

**Pull request closed without merging**